concourse/steps/version.mako

```diff
@@ -21,13 +22,22 @@ path_to_repo_version_file = os.path.join(
 output_version_file = os.path.join(job_step.output('version_path'), 'version')
 legacy_version_file = os.path.join(job_step.output('version_path'), 'number')
 
-if (read_callback := version_trait.read_callback()):
+# Assign empty string to calbacks if None, as we'd template 'None' (the string) later otherwise
+if (read_callback := version_trait.read_callback() or ''):
```

Comment:
you might initialise `read_callback` w/ empty str as initial value; otoh: why actually default to empty str? None should be okay as a marker for "not set"

Reply:
I only introduced the same check that was already existing for the `write_callback` also for the `read_callback`. Do you want me to change how this is done for both? My understanding was that we're specifically trying to avoid having the literal string `None` in the rendered pipeline.

Reply:
Of course, it would be nicer to not render the str-literal `None` into the template. However, I think this might be done even better by not rendering that part at all by guarding the line(s) in question with a
```
% if <check for callback>
# do stuff with callback
% endif
```

That way, there would also not be a str-literal `None` rendered into the code

Reply:
I think I understand what you want this to look like, but couldn't do it easily without re-structuring `version.mako` quite a bit. 
* There's quite a few assumptions that e.g. `write_callback` always exists as a variable
* It feels weird to me to access the callbacks based on `version_interface: callback` in `version.mako`, but you cannot see why it could never happen that this is set, but the callback variables are undefined

So for this PR, I wouldn't include that change, if that's ok with you?